src/Core/Extensible.php

```diff
@@ -553,8 +557,21 @@ public function getExtensionInstances()
 
             if ($extensions) {
                 foreach ($extensions as $extension) {
-                    $instance = Injector::inst()->get($extension);
-                    $this->extension_instances[get_class($instance)] = $instance;
+                    $name = $extension;
+                    // Allow service names of the form "%$ServiceName"
+                    if (substr($name, 0, 2) == '%$') {
+                        $name = substr($name, 2);
```

Comment:
`$name = ltrim($name, '%$');` faster :)

Reply:
`ltrim($name, '%$')` would also turn strings starting with `$%` or `$` or `%` or multiple combinations of those as the second argument to `trim` functions is a list of chars to trim.

Reply:
Yeah and none of those characters are in a class name :)